WORKSPACE

```diff
@@ -12,6 +13,56 @@ load("//:setup.bzl", "kythe_rule_repositories")
 
 kythe_rule_repositories()
 
+# Tweag Nix Support
+######################
+
+# Replace with http_archive: https://github.com/tweag/rules_nixpkgs/#setup
+_commit = "c061458ec4c0e2bfc373b3aa7c804708ebee5474"
+http_archive(
+    name = "io_tweag_rules_nixpkgs",
+    strip_prefix = "rules_nixpkgs-{}".format(_commit),
+    urls = ["https://github.com/tweag/rules_nixpkgs/archive/{}.tar.gz".format(_commit)],
+    sha256 = "2e39a8d0b373b0427c85bc20f144fcf1f4b0f9d1504d0d0455e00192c0cb2c27",
+)
+
+load("@io_tweag_rules_nixpkgs//nixpkgs:repositories.bzl", "rules_nixpkgs_dependencies")
+rules_nixpkgs_dependencies()
+
+# Define nixpkgs version using the flakes lock file.
+load("@io_tweag_rules_nixpkgs//nixpkgs:nixpkgs.bzl", "nixpkgs_local_repository")
+nixpkgs_local_repository(
+    name = "nixpkgs",
+    nix_flake_lock_file = "//:flake.lock",
+    nix_file_deps = ["//:flake.lock"],
+)
+
+# Configure the C++ toolchain
+load("@io_tweag_rules_nixpkgs//nixpkgs:nixpkgs.bzl", "nixpkgs_cc_configure")
+nixpkgs_cc_configure(
+    name = "nixpkgs_config_cc",
+    repository = "@nixpkgs",
+    attribute_path = "clang_14",
+    cc_lang = "c++",
+    cc_std = "c++11",
+)
+
+# Configure Java toolchain.
+load("@io_tweag_rules_nixpkgs//nixpkgs:nixpkgs.bzl", "nixpkgs_java_configure")
+nixpkgs_java_configure(
+    attribute_path = "jdk17.home",
+    repository = "@nixpkgs",
+    toolchain = True,
+    toolchain_name = "nixpkgs_java",
+    toolchain_version = "17",
```

Comment:
This will definitely need to support a newer Java version as Google was already using at least Java 20 when I left.

Reply:
Noted. I think we can go up to jdk22, but I haven't tried it. Getting all this to work was a challenge enough.

Reply:
I think that so long as nix supports jdk17, there should be no need to worry.  You need to worry if you depend on the JDK available on the target, but since we install our own hermetic one, it's actually a non-issue.

In other news, I wasn't able to get any other JDK to install hermetically. I'll see what is up about that.

Reply:
Ok, now using JDK20.